metrics.go

```diff
@@ -103,45 +109,64 @@ func (c *aggregatingCollector) Describe(descs chan<- *prometheus.Desc) {
 	}
 }
 
-func (c *aggregatingCollector) Collect(metrics chan<- prometheus.Metric) {
-	now := time.Now()
+func (c *aggregatingCollector) cron() {
+	ticker := time.NewTicker(collectFrequency)
```

Comment:
We need some way to stop this or we're going to mess with our tests.

Can we implement `Transport.Close`?

We could also auto-close this when the last connection dies, but that's slightly tricker.

Reply:
How is this messing with our tests?
The `collector` is a global variable, and it's initialized in `init()`. On the other hand, you can have multiple transports.

Reply:
Oh, nvm. I thought this was per-transport. In that, no, it'll be fine.